### PR TITLE
New vehicle part: back-up alarm

### PR DESCRIPTION
**Pull request by illi-kun**
This device can bring some unexpected troubles to unprepared player.

It is installed on the following vehicles: 
- Schoolbus
- RV
- Luxury RV
- Security Van
- Semi Truck
- Ambulance
- Fire Truck
- Fire Engine
- Hippie Van
- Low-End Cube Van
- Cube Van
- Flatbed Truck

As usual, I'd appreciate any editor's help with item description.

### CODE REVIEW

File: data/json/items/vehicle_parts.json [hunk 1]
```diff
@@ -917,6 +917,22 @@
         "category" : "veh_parts",
         "price": 80000
     },
+   {
+        "type":"GENERIC",
+        "id": "beeper",
+        "name": "back-up beeper",
+        "description": "This is a safety device intended to warn passersby of a vehicle moving in reverse, but the usage of it now looks terribly unsafe.",
```
Comment (DavidKeaton) [inline data/json/items/vehicle_parts.json]: "now seems terribly unwise." IMO. you don't see an alarm. :-)

Comment (illi-kun) [inline data/json/items/vehicle_parts.json]: Accepted, thanks!

File: src/vehicle.cpp [hunk 2]
```diff
@@ -1433,6 +1433,31 @@ void vehicle::honk_horn()
     }
 }
 
+void vehicle::beeper_sound()
+{
+    // No power = no sound
+    if( !fuel_left( fuel_type_battery, true ) ) {
```
Comment (Coolthulhu) [inline src/vehicle.cpp]: Code quality nitpick: comparison with 0 is preferable to implicit cast to bool

File: data/json/vehicle_parts.json [hunk 3]
```diff
@@ -4000,6 +4000,23 @@
...
+        "broken_symbol" : "*",
+        "broken_color" : "dark_gray",
+        "damage_modifier" : 10,
+        "durability" : 90,
+        "bonus" : 28,
+        "item" : "beeper",
+        "difficulty" : 2,
+        "flags" : ["BEEPER", "ODDTURN"],
+        "breaks_into" : [
+            {"item": "steel_chunk", "min": 0, "max": 1}
```
Comment (Coolthulhu) [inline data/json/vehicle_parts.json]: Some electronics would be nice here. Amplifiers sound like something to be expected from things that make sound.

**Comment by DavidKeaton** [comment]
np bro :-)

File: data/json/vehicle_parts.json (hunk 3 above)
Comment (DavidKeaton) [inline data/json/vehicle_parts.json]: and, as per usual, copper wires. :-)

Comment (illi-kun) [inline data/json/vehicle_parts.json]: I'll add wires, but extraction of amplifier in pristine condition by breakage a beeper seems weird to me.

Comment (DavidKeaton) [inline data/json/vehicle_parts.json]: point, how about electronic scraps?

Comment (illi-kun) [inline data/json/vehicle_parts.json]: good point, will add it too.

**Comment by BigD145** [comment]
Hippie Vans have those? Since when?

**Comment by DavidKeaton** [comment]
since this was merged lol

**Comment by Rivet-the-Zombie** [comment]
I've never heard of a Microbus with a backup alarm, but this is supposed to be happening in the 2040s so it's possible.

**Comment by DavidKeaton** [comment]
even people have alarms in the future!

**Comment by illi-kun** [comment]
I've used this phrase from Wiki as reference: "Back-up beepers or an observer are required by OSHA for construction equipment with an obstructed view to the reverse". It's applicable to Hippie Van, take a look: 
view

I can revert this part of PR, if it's important.

**Comment by Rivet-the-Zombie** [comment]
I've never actually seen a VW Microbus (as I'm assuming that's what our 'Hippie Van' is supposed to represent) used for any commercial purpose, so I wouldn't think OSHA would have much authority over such things.

But as I said before, DDA takes place in the 2040s, so things may have changed in the interim.

**Comment by DavidKeaton** [comment]
Regardless, nice PR. I love vehicles having that much more savory flavour.

**Pull request merged**